mod_auth_matrix_user_verification.lua

```diff
@@ -124,12 +224,14 @@ local function process_and_verify_token(session)
         return false, "access-denied", "Jitsi room does not match Matrix room"
     end
 
-    local result = verify_room_membership(data.context.matrix)
+    local isMember, isOwner = verify_room_membership(data.context.matrix)
 
-    if result == false then
-    return false, "access-denied", "Token invalid or not in room";
+    if isMember == false then
+        return false, "access-denied", "Token invalid or not in room";
     end
 
+    -- Store the isOwner detail
+    session.auth_matrix_user_verification_is_owner = isOwner
```

Comment:
hmm, may be worth adding the room name either as a separate variable or to the name of this one? Otherwise I wonder if it would be possible to auth yourself against one matrix room but then gain moderator status in a different jitsi conference, somehow.

Reply:
Hmm, I'm not entirely sure. I was thinking of adding the Jitsi room name, but it's already in the session. And using a session for some room against another room seems unlikely to be possible, considering these are internal structures according to the Prosody team? 

There is this above: `    if not (target_room_name == session.jitsi_meet_room) then` which if the check failes does not owner the user in the room. I'm not entirely sure why that is there, I took it from the Jitsi Meet plugin that admins everyone (https://github.com/jitsi/jitsi-meet/blob/master/resources/prosody-plugins/mod_muc_allowners.lua#L63), but it's possible it's something to do with ensuring the session is for this particular room. If that is the case, it will also protect in this case I believe.

Adding the room name here and checking it in the later phase could be done but it feels like that is already happening. We don't have the matrix room ID in the "participant joined" hook, so we can't really compare to that, and the jitsi room name is already being compared to the one in the session.

Reply:
ah ok, if that jitsi room name in the session is defined to be the one matching where the power levels come from then yeah, it should be fine